### ISSUE TITLE
About the (unjustified) takedown

### ISSUE THREAD

**Issue by blackjack4494** [blackjack4494/YDLC#2]
I am not really sure whether that DMCA would actually hold in court whether USA and especially in Germany/EU. Apparently the reason was that some VEVO videos were used for testing purposes.  
The takedown notice let it sound like the sole purpose of youtube-dl is to download only copyrighted content specifically VEVO. But it's a universal downloader that allows to download ALL videos on youtube.  
Furthermore the project supports sooo many more sites other than youtube. It's in no means justified to take it down. What would have been justified is to force remove the code that is handling youtube.

**Comment by mon5termatt** [comment]
yeah its invalid, its ultimately just utter bullshit. I'm sure it will be back up within a few days once they remove the hard coded links, that way there is no direct issues.

**Comment by mjb2010** [comment]
There are two ways to issue a DMCA takedown. One is the way most people are familiar with, in which the claimant must send a special form letter with all the right language, "under penalty of perjury", etc.; and then the online service provider is _required_ to remove the offending material.

The other is to simply provide what is called in U.S. law "red flag knowledge" of illegality. See p. 113 of the May 2020 report _[Section 512 of Title 117](https://www.copyright.gov/policy/section512/section-512-full-report.pdf)_ ("Actual Knowledge vs. Red Flag Knowledge"). Providing this kind of notice requires a a much less formal letter which need not refer to specific acts of infringement. It can just point out that something under the OSP's purview is a conduit/venue/tool for specific acts of infringement, if not being infringing, itself. The OSP is not required to remove the material, but if they don't, they lose their DMCA safe harbor...so it is effectively as good as a regular DMCA takedown notice. In fact, it's more effective because it eliminates the entire counternotice/reinstatement process.

This "red flag knowledge" standard has been in effect since the Napster debacle. It was slightly curtailed by the Viacom case but overall it remains in effect. Microsoft/GitHub is not going to risk a court battle over youtube-dl(c) anyway.

One might argue that there are non-infringing uses of the software, but the same thing was true of Napster. They were sunk, in part, by the fact that they marketed it as having infringing uses. In U.S. civil court, that was sufficient to incur liability.

So I don't think trying to split hairs over the legitimacy of the letter will get you very far. Eliminating any hint of copyright infringement/circumvention purposes and examples from the core program, removing YouTube from its name, and having the contentious functionality be in code hosted elsewhere is the better option.

**Comment by basxto** [comment]
> removing YouTube from its name

Downloading from youtube isn’t per se a copyright infringement.
Youtube allows to put videos under creative commons licenses: https://www.youtube.com/c/DroidconDe/videos
But it is indeed wise to remove it because of trademarks etc.
